LeaderboardBackend/Services/Impl/CategoryService.cs

```diff
@@ -17,6 +17,23 @@ public class CategoryService(ApplicationContext applicationContext, IClock clock
         await applicationContext.Categories
             .FirstOrDefaultAsync(c => c.Slug == slug && c.LeaderboardId == leaderboardId && c.DeletedAt == null);
 
+    public async Task<GetCategoriesForLeaderboardResult> GetCategoriesForLeaderboard(long leaderboardId, bool includeDeleted)
+    {
+        Leaderboard? board = await applicationContext.Leaderboards
+            .Where(board => board.Id == leaderboardId)
+            .Include(board => board.Categories)
+            .FirstOrDefaultAsync();
```

Comment:
This is an incredibly minor nitpick but use `SingleOrDefaultAsync()` instead.

Reply:
Happy to change it, but what's the difference?